Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2539 @@
+            }
+        }
+    }
+    
+    /*
+     The color used to tint the thumb's border.
+     */
+    public var playbackButtonsFillColor: UIColor = UIColor.black {
+        didSet {
+            //
+            // Set new button images created with new fill color
+            let playPauseButtonImage = isPlaying ? pauseImage?.imageWithColor(playbackButtonsFillColor) : playImage?.imageWithColor(playbackButtonsFillColor)
+            playPauseButton?.setBackgroundImage(playPauseButtonImage, for: .normal)
+            forwardButton?.setBackgroundImage(forwardImage?.imageWithColor(playbackButtonsFillColor), for: .normal)
+            backButton?.setBackgroundImage(backImage?.imageWithColor(playbackButtonsFillColor), for: .normal)
+        }
+    }
+    
+    /*
+     Playback direction when time is being animated.
+     */
+    public var playbackDirection: PlaybackDirection = .forward
+    
+    /*
+     The looping behavior of the slider when time is being animated.
+     */
+    public var playbackLoopMode: LoopMode = .repeat
+    
+    /*
+     The amount of time during playback that will elapse before the slider advances
+     to the next time step. If geoView is available then playback will wait until
+     geoView's drawing (drawStatus) is completed before advances to the next time step.
+     */
+    public var playbackInterval: TimeInterval = 1
+    
+    /*
+     A Boolean value indicating whether playback is currently running.
+     */
+    public var isPlaying: Bool = false {
+        didSet {
+            if isPlaying {
+                //
+                // Set the pause image
+                playPauseButton?.setBackgroundImage(pauseImage?.imageWithColor(playbackButtonsFillColor), for: .normal)
+                
+                // Invalidate timer, in case this button is tapped multiple times
+                timer.invalidate()
+                
+                // Start the timer with specified playback interval
+                timer = Timer.scheduledTimer(timeInterval: playbackInterval, target: self, selector: #selector(timerAction), userInfo: nil, repeats: true)
+            }
+            else {
+                //
+                // Set the play image
+                playPauseButton?.setBackgroundImage(playImage?.imageWithColor(playbackButtonsFillColor), for: .normal)
+                
+                // Invalidate timer.
+                timer.invalidate()
+            }
+        }
+    }
+    
+    // MARK: Other Properties
+    
+    /*
+     This will initialize the slider's fullExtent, currentExtent and timeStepInterval
+     properties based on time enabled layers in the geo view.
+     */
+    public private(set) var geoView: AGSGeoView?
+    
+    /*
+     A Boolean value indicating whether slider is visible or not. Default is true.
+     */
+    public var isSliderVisible: Bool = true {
+        didSet {
+            layoutIfNeeded()
+            UIView.animate(withDuration: 0.25) {
```

Comment:
This animation seems unnecessary because the frame is adjusted without animation and thus the components that are becoming animated hidden are doing so after the frame has adjusted and thus looks strange. Consider wrapping the frame adjustment as an animation within this block as well or removing this animation block altogether.

Reply:
same comment as above.